### ISSUE TITLE
Add a template tag to use already defined queries

### ISSUE THREAD

**Issue by salsakran** [metabase/metabase#5579]
At its simplest, it could be something like

`with {{card 12}} as foo`

Ideally you could search for cards by name as well. Not sure how best to do this given the constraints of how template tags are found + parsed right this moment. 

Thoughts @mazameli , @camsaul or @senior ?

**Comment by mazameli** [comment]
IDK if we'd be able to do this currently, but it could be cool if we see the string `with {{` then we'd show a list of saved questions like how we do with tables etc. Problem with that is it could be a super long list, so we'd have to be smart about a minimum number of characters you have to type before we show you the list.

![screen shot 2017-07-25 at 4 35 05 pm](https://user-images.githubusercontent.com/2223916/28598271-6135df5a-7157-11e7-9fd7-5e709717859c.png)

Falling short of that, I _guess_ you could put in a top-right button (or add a Native action?) to `insert query reference` or whatever.

**Comment by ebalgava** [comment]
This feature could save me up a lot of time because of maintenance and readability of nested queries.

**Comment by mirceapasoi** [comment]
Same here, it would be super useful to us!

**Comment by mfrasco6** [comment]
I fully support this feature as an effort to minimize the number of times the same sub-query is used in multiple questions.

**Comment by a-warner** [comment]
this would save us a ton of time, would love to see this feature in a future version!

**Comment by AlexHuckert** [comment]
We would love that. Any updates on this topic ?

**Comment by devxoul** [comment]
It would be great if we could pass the variables to the nested questions:

```sql
-- use the nested question as a subquery
WITH {% question 13, variable1=foo, variable2=bar %} AS data
SELECT ...
```

```sql
-- use the nested question as a complete query itself with some variables
{% question 13, variable1={{ var_foo }}, variable2=const_bar %}
```

**Comment by trakhov** [comment]
It would be extremely useful. Any updates on the topic?

**Comment by sanchexx** [comment]
Likewise, it would be useful, not only for maintenance, but also for performance.

I have a metabase dashboard with 20 questions. Every question use the same subquery. Right now I copy paste this subquery between all SQL, which makes it hard to maintain, but also less performant.

By splitting this subquery, it would run once, cache the result, and then make all other queries much faster. Unless anyone have a better idea around that?

**Comment by karthik-vv** [comment]
This feature is present in "Custom question" were we can select "Saved Questions" where it asks for picking the data. 
But same thing in "Native query" will be extremely useful.

**Comment by flamber** [comment]
:wave: This was added by PR #11835 (#11804) and will be available in the upcoming 0.35.0 :tada:
